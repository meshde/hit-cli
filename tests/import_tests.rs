mod fixtures;
use assert_cmd::prelude::*;
use fixtures::{get_hit_command_for_dir, temp_dir};
use rstest::*;
use std::fs;
use std::path::PathBuf;
use tempfile::TempDir;

#[rstest]
fn test_import_swagger(temp_dir: TempDir) {
    fs::copy(
        PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("tests/fixtures/swagger.yml"),
        temp_dir.path().join("swagger.yml"),
    )
    .unwrap();

    let mut cmd = get_hit_command_for_dir(&temp_dir.path());
    cmd.args(["import", "./swagger.yml"]);
    cmd.assert().success();

    let config_path = temp_dir.path().join(".hit").join("config.json");
    let reader = fs::File::open(config_path).unwrap();
    let hit_config: serde_json::Value = serde_json::from_reader(reader).unwrap();
    insta::assert_json_snapshot!(hit_config);
}
