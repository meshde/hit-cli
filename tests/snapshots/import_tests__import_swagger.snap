---
source: tests/import_tests.rs
expression: hit_config
---
{
  "commands": {
    "pet": {
      "add-pet": {
        "body": {
          "category": {},
          "id": 0,
          "name": "",
          "photoUrls": [],
          "status": "",
          "tags": []
        },
        "headers": {},
        "method": "POST",
        "postscript": null,
        "url": "{{API_URL}}/pet"
      },
      "delete-pet": {
        "body": null,
        "headers": {},
        "method": "DELETE",
        "postscript": null,
        "url": "{{API_URL}}/pet/:petId"
      },
      "find-pets-by-status": {
        "body": null,
        "headers": {},
        "method": "GET",
        "postscript": null,
        "url": "{{API_URL}}/pet/findByStatus?status=:status"
      },
      "find-pets-by-tags": {
        "body": null,
        "headers": {},
        "method": "GET",
        "postscript": null,
        "url": "{{API_URL}}/pet/findByTags?tags=:tags"
      },
      "get-pet-by-id": {
        "body": null,
        "headers": {},
        "method": "GET",
        "postscript": null,
        "url": "{{API_URL}}/pet/:petId"
      },
      "update-pet": {
        "body": {
          "category": {},
          "id": 0,
          "name": "",
          "photoUrls": [],
          "status": "",
          "tags": []
        },
        "headers": {},
        "method": "PUT",
        "postscript": null,
        "url": "{{API_URL}}/pet"
      },
      "update-pet-with-form": {
        "body": null,
        "headers": {},
        "method": "PUT",
        "postscript": null,
        "url": "{{API_URL}}/pet/:petId?name=:name&status=:status"
      },
      "upload-file": {
        "body": null,
        "headers": {},
        "method": "GET",
        "postscript": null,
        "url": "{{API_URL}}/pet/:petId/uploadImage?additionalMetadata=:additionalMetadata"
      }
    },
    "store": {
      "delete-order": {
        "body": null,
        "headers": {},
        "method": "DELETE",
        "postscript": null,
        "url": "{{API_URL}}/store/order/:orderId"
      },
      "get-inventory": {
        "body": null,
        "headers": {},
        "method": "GET",
        "postscript": null,
        "url": "{{API_URL}}/store/inventory"
      },
      "get-order-by-id": {
        "body": null,
        "headers": {},
        "method": "GET",
        "postscript": null,
        "url": "{{API_URL}}/store/order/:orderId"
      },
      "place-order": {
        "body": {
          "complete": false,
          "id": 0,
          "petId": 0,
          "quantity": 0,
          "shipDate": "",
          "status": ""
        },
        "headers": {},
        "method": "GET",
        "postscript": null,
        "url": "{{API_URL}}/store/order"
      }
    },
    "user": {
      "create-user": {
        "body": {
          "email": "",
          "firstName": "",
          "id": 0,
          "lastName": "",
          "password": "",
          "phone": "",
          "userStatus": 0,
          "username": ""
        },
        "headers": {},
        "method": "POST",
        "postscript": null,
        "url": "{{API_URL}}/user"
      },
      "create-users-with-list-input": {
        "body": {},
        "headers": {},
        "method": "POST",
        "postscript": null,
        "url": "{{API_URL}}/user/createWithList"
      },
      "delete-user": {
        "body": null,
        "headers": {},
        "method": "DELETE",
        "postscript": null,
        "url": "{{API_URL}}/user/:username"
      },
      "get-user-by-name": {
        "body": null,
        "headers": {},
        "method": "GET",
        "postscript": null,
        "url": "{{API_URL}}/user/:username"
      },
      "login-user": {
        "body": null,
        "headers": {},
        "method": "GET",
        "postscript": null,
        "url": "{{API_URL}}/user/login?username=:username&password=:password"
      },
      "logout-user": {
        "body": null,
        "headers": {},
        "method": "GET",
        "postscript": null,
        "url": "{{API_URL}}/user/logout"
      },
      "update-user": {
        "body": {
          "email": "",
          "firstName": "",
          "id": 0,
          "lastName": "",
          "password": "",
          "phone": "",
          "userStatus": 0,
          "username": ""
        },
        "headers": {},
        "method": "PUT",
        "postscript": null,
        "url": "{{API_URL}}/user/:username"
      }
    }
  },
  "envs": {
    "prod": {
      "API_URL": "https://petstore3.swagger.io/api/v3"
    }
  }
}
